src/plugins/dashboard/server/usage/dashboard_telemetry.ts

```diff
@@ -115,6 +119,23 @@ export const collectForPanels: DashboardCollectorFunction = (panels, collectorDa
   collectByValueLensInfo(panels, collectorData);
 };
 
+export const collectEmbeddableData = (
+  panels: SavedDashboardPanel730ToLatest[],
+  collectorData: DashboardCollectorData,
+  embeddableService: EmbeddablePersistableStateService
+) => {
+  for (const panel of panels) {
+    collectorData.embeddable = embeddableService.telemetry(
```

Comment:
Just so I understand, `embeddableService.telemetry` would mutate `collectorData.embeddable` on each iteration so in the end, we get the resulting object with telemetry from each panel? 

Hard to track internals of `embeddableService.telemetry`

Reply:
Yes, you are right, with the exception that `embeddableService.telemetry` function will not mutate `collectorData.embeddable` but return a "copy" of it.